src/psyclone/psyir/backend/fortran.py

```diff
@@ -704,11 +704,12 @@ def container_node(self, node):
             subroutines += self._visit(child)
 
         result += (
-            "{1}\n"
+            "{1}"
             "{0}implicit none\n"
-            "{0}contains\n"
             "{2}\n"
-            "".format(self._nindent, declarations, subroutines))
+            "{0}contains\n"
```

Comment:
examples/gocean/eg2 now fails to compile because the generated kernel code has two `CONTAINS` in it.

Reply:
Is was actually a bug in the backend, the 2 'contains' are right. I added a specific test for it now.